src/coll/options.rs

```diff
@@ -622,6 +622,10 @@ pub struct AggregateOptions {
 #[builder(field_defaults(default, setter(into)))]
 #[non_exhaustive]
 pub struct CountOptions {
+    /// The filter to use for the operation.
+    #[serde(skip_serializing)]
+    pub filter: Option<Document>,
```

Comment:
Is the plan to move `filter` into options for all methods that accept one? This is at odds with the definitions in the CRUD spec, which have `filter` as a required parameter adjacent to the options. I'm not sure I have strong feelings about which design is better, but we might want to make sure there isn't a strong reason the CRUD definitions were written this way before deviating. (edit: looks like we already deviate by accepting filters as optional docs in our existing API...)

Reply:
good point about silent defaults, sgtm

Reply:
So it turns out that the spec has `filter` as an explicit non-optional parameter for [`countDocuments`](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#L171), but as just another option for [`listDatabases`](https://github.com/mongodb/specifications/blob/master/source/enumerate-databases.rst#filters) and [`listCollections`](https://github.com/mongodb/specifications/blob/master/source/enumerate-collections.rst#post-mongodb-280-rc3-versions).  I'm going to stick to mirroring that for the Rust driver; happy to do a follow-up PR if you think it'd be better to be non-optional for all three.

Reply:
yeah going by the spec seems like the right approach here + for the other methods that accept a filter in future PRs